File: examples/fsm_platformer/fsm_platformer_python/player_falling_state.py
"""Player Falling State — downward arc (vy <= 0)."""

from __future__ import annotations

from src.engine.core import MonoBehaviour
from src.engine.math.vector import Vector2

from fsm_platformer_python.fsm import FSMState


class PlayerFallingState(FSMState):
    """Player is falling downward."""

    def act(self, owner: MonoBehaviour) -> None:
        from fsm_platformer_python.player_input_handler import PlayerInputHandler
        player: PlayerInputHandler = owner  # type: ignore

        # Allow horizontal control while falling
        rb = player.rb
        h = player.horizontal_input
        rb.velocity = Vector2(h * player.move_speed, rb.velocity.y)

File: examples/fsm_platformer/fsm_platformer_unity/PlayerFallingState.cs
using UnityEngine;

// Unity APIs used: MonoBehaviour, Rigidbody2D, Vector2
public class PlayerFallingState : FSMState
{
    public override void Act(MonoBehaviour owner)
    {
        PlayerInputHandler player = (PlayerInputHandler)owner;
        float h = player.HorizontalInput;
        Rigidbody2D rb = player.Rb;
        rb.linearVelocity = new Vector2(h * player.MoveSpeed, rb.linearVelocity.y);
    }
}
